hazelcast/sql.py

```diff
@@ -609,6 +626,12 @@ def metadata(self):
         """SqlRowMetadata: The row metadata."""
         return self._row_metadata
 
+    def __getitem__(self, item):
+        if isinstance(item, six.integer_types):
+            return self.get_object_with_index(item)
+
+        return self.get_object(item)
```

Comment:
Just to conform to what python docs says: 


> Called to implement evaluation of self[key]. For sequence types, the accepted keys should be integers and slice objects. Note that the special interpretation of negative indexes (if the class wishes to emulate a sequence type) is up to the __getitem__() method. If key is of an inappropriate type, TypeError may be raised; if of a value outside the set of indexes for the sequence (after any special interpretation of negative values), IndexError should be raised. For mapping types, if key is missing (not in the container), KeyError should be raised.


at
https://docs.python.org/3/reference/datamodel.html#object.__getitem__

Also note that we raise `ValueError` if key is a string and there is no column with that name.  According to documentation we should raise a `KeyError`.

The following conforms to the documentation:

```suggestion
        if isinstance(item, six.integer_types):
            return self.get_object_with_index(item)

        if isinstance(item, six.string_types):
            index = self._row_metadata.find_column(column_name)
            if index == SqlRowMetadata.COLUMN_NOT_FOUND:
                raise KeyError("Column '%s' doesn't exist" % column_name)
            return self.get_object(item)

        raise TypeError
```

Reply:
I feel like it is okay to raise `ValueError` here. As our documentation suggests, this is just an easier way to use `get_object` and `get_object_with_index`.